### ISSUE TITLE
Tile size cannot exceed image size

### ISSUE THREAD

**Issue by bencost97** [acl21/image_bbox_slicer#4]
Hi,

I am getting the above error due to the image size being (0,0). I have a JPEG image in the src/image folder but for some reason the image slicer is not reading it properly. How do I get the slicer to properly read this?

Thanks!

**Comment by tntrela** [comment]
Hello, 
I am facing the same issue. 
Could you let me know how you fixed it?
Thanks in advance

**Comment by acl21** [comment]
Hello @tntrela, sorry for the delayed response. I am currently busy with my college coursework. Are you still facing the issue? If yes, could you please post the full error message here? I will try my best to find out what could be causing the problem. Good day!  

P.S. I am opening the issue.

**Comment by tntrela** [comment]
Hello @acl21, thank you for spending the time to have a look at it.
 Just a note I checked and the images open just fine when using just PIL. 

ValueError                                Traceback (most recent call last)
<ipython-input-32-a39f5e33444e> in <module>
----> 1 slicer.slice_by_number(number_tiles=3)

~/.local/lib/python3.8/site-packages/image_bbox_slicer/main.py in slice_by_number(self, number_tiles)
    166         """
    167         self._just_image_call = False
--> 168         self.slice_bboxes_by_number(number_tiles)
    169         self.slice_images_by_number(number_tiles)
    170         self._ignore_tiles = []

~/.local/lib/python3.8/site-packages/image_bbox_slicer/main.py in slice_bboxes_by_number(self, number_tiles)
    288         validate_number_tiles(number_tiles)
    289         self._ignore_tiles = []
--> 290         mapper = self.__slice_bboxes(None, None, number_tiles=number_tiles)
    291         if self.save_before_after_map:
    292             save_before_after_map_csv(mapper, self.ANN_DST)

~/.local/lib/python3.8/site-packages/image_bbox_slicer/main.py in __slice_bboxes(self, tile_size, tile_overlap, number_tiles)
    314             else:
    315                 tile_w, tile_h = tile_size
--> 316             tiles = self.__get_tiles((im_w, im_h), tile_size, tile_overlap)
    317             tile_ids = []
    318 

~/.local/lib/python3.8/site-packages/image_bbox_slicer/main.py in __get_tiles(self, img_size, tile_size, tile_overlap)
    119             in the format - `(xmin, ymin, xmax, ymax)`
    120         """
--> 121         validate_tile_size(tile_size, img_size)
    122         tiles = []
    123         img_w, img_h = img_size

~/.local/lib/python3.8/site-packages/image_bbox_slicer/helpers.py in validate_tile_size(tile_size, img_size)
    210         if isinstance(img_size, tuple):
    211             if (sum(tile_size) >= sum(img_size)) or (tile_size[0] > img_size[0]) or (tile_size[1] > img_size[1]):
--> 212                 raise ValueError('Tile size cannot exceed image size. Tile size was {} while image size was {}'.format(
    213                     tile_size, img_size))
    214         else:

ValueError: Tile size cannot exceed image size. Tile size was (0, 0) while image size was (0, 0)

**Comment by acl21** [comment]
Thank you @tntrela for the error message. It helps a lot. The image size is (0,0) so it must be a file reading issue. Is it possible for you to share the image? If not, just it's properties? 

Thank you in advance!

**Comment by tntrela** [comment]
![225399565_417971036133710_6320600321498764376_n](https://user-images.githubusercontent.com/42609260/147453403-00b55227-0243-4746-9b05-bb0d76fc2432.png)
It is just a normal png image.

**Comment by acl21** [comment]
Thank you for the image @tntrela. Could you also share the corresponding annotation file? 

Sorry for the trouble, I am unable to spot the problem. So I am doing everything I can to pinpoint the bug. Thanks in advance! 😃

**Comment by tntrela** [comment]
[Example.zip](https://github.com/acl21/image_bbox_slicer/files/7791720/Example.zip)
Hello again, here you go. Thanks for all the effort!

**Comment by acl21** [comment]
@tntrela So I found the problem. The annotation file you shared doesn't have the right image size (also doesn't have the right annotations just to be clear). Precisely, it has image width and height as 0. Hence the "image size was (0, 0)" error.

```xml
<size>
    <width>0</width>
    <height>0</height>
    <depth>3</depth>
</size>
```  

I cannot bypass the existing slicing logic around this because the functions I wrote must work independently i.e., one should be able to slice images and annotations separately without any codependence. So the only way to fix it is to make sure your annotation files have the right corresponding image sizes. 😕 

I hope this helps. Good luck! 👍
